tiledb-0.5.6 python packages fail a test and can't be build

**Describe the bug**
Same as #80018, since tests started failing again without changes to the package.

Tests for the tiledb python-modules fail. The failing tests are not always the same. I saw a least 3 different tests with the same package.
Looking at https://hydra.nixos.org/job/nixpkgs/trunk/python37Packages.tiledb.x86_64-linux the 3 successful builds on hydra do not correspond with changes to the package (python module or native).

@FRidh Maybe the offending tests should be disabled? I also tried 0.5.9 and also had failing tests.

**To Reproduce**
Steps to reproduce the behavior:
1. build python37Packages.tiledb on master (from nixpkgs-directory `nix-build . -A python37Packages.tiledb`)

**Expected behavior**
Package should build.

**Screenshots**
```
======================================================================
ERROR: test_ctx_thread_cleanup (test_libtiledb.HighlevelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/sj4r23dx840n2ih9d57qb7w9w0k6a6lr-python3.7-psutil-5.6.7/lib/python3.7/site-packages/psutil/_pslinux.py", line 1517, in wrapper
    return fun(self, *args, **kwargs)
  File "/nix/store/sj4r23dx840n2ih9d57qb7w9w0k6a6lr-python3.7-psutil-5.6.7/lib/python3.7/site-packages/psutil/_pslinux.py", line 1905, in threads
    st = f.read().strip()
ProcessLookupError: [Errno 3] No such process

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/1wfcpkqzgviv258a3ykm76wrw0fbll3y-python3.7-tiledb-0.5.6/lib/python3.7/site-packages/tiledb/tests/test_libtiledb.py", line 2507, in test_ctx_thread_cleanup
    start_threads = len(thisproc.threads())
  File "/nix/store/sj4r23dx840n2ih9d57qb7w9w0k6a6lr-python3.7-psutil-5.6.7/lib/python3.7/site-packages/psutil/__init__.py", line 987, in threads
    return self._proc.threads()
  File "/nix/store/sj4r23dx840n2ih9d57qb7w9w0k6a6lr-python3.7-psutil-5.6.7/lib/python3.7/site-packages/psutil/_pslinux.py", line 1521, in wrapper
    raise NoSuchProcess(self.pid, self._name)
psutil.NoSuchProcess: psutil.NoSuchProcess process no longer exists (pid=79)

======================================================================
FAIL: test_metadata_consecutive (test_metadata.MetadataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/1wfcpkqzgviv258a3ykm76wrw0fbll3y-python3.7-tiledb-0.5.6/lib/python3.7/site-packages/tiledb/tests/test_metadata.py", line 147, in test_metadata_consecutive
    self.assertEqual(A.meta['randint'], randints[-1])
AssertionError: 5809793530899818813 != 9105086306405040236

----------------------------------------------------------------------
Ran 138 tests in 133.157s
```
```
======================================================================
FAIL: test_varlen_write_int_subarray (test_libtiledb.DenseVarlen)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/g8in2avxx1gp3zca869xxz9zlcgj1r9q-python3.7-tiledb-0.5.6/lib/python3.7/site-packages/tiledb/tests/test_libtiledb.py", line 1310, in test_varlen_write_int_subarray
    assert_subarrays_equal(A_assigned, T[:])
  File "/nix/store/g8in2avxx1gp3zca869xxz9zlcgj1r9q-python3.7-tiledb-0.5.6/lib/python3.7/site-packages/tiledb/tests/common.py", line 47, in assert_subarrays_equal
    assert_array_equal(a_el, b_el)
  File "/nix/store/v6rdicqjk67y15s7pskn0clm4wy87asb-python3.7-numpy-1.18.1/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 936, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/nix/store/v6rdicqjk67y15s7pskn0clm4wy87asb-python3.7-numpy-1.18.1/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 765, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Arrays are not equal

(shapes (6,), (2,) mismatch)
 x: array([ 5,  6,  7,  8,  9, 10], dtype=uint64)
 y: array([1, 2], dtype=uint64)
```
**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.5.9, NixOS, 20.09pre217537.d96bd3394b7 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(holger): `""`
 - channels(root): `"nixos-20.09pre217537.d96bd3394b7, old-18.09.2574.a7e559a5504, unstable-20.09pre214294.1ddb140d95c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - python37Packages.tiledb
  - python38Packages.tiledb
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/tiledb/default.nix
+++ b/pkgs/development/python-modules/tiledb/default.nix
@@ -51,10 +51,15 @@ buildPythonPackage rec {
     # Hardcode path to shared object
     substituteInPlace tiledb/__init__.py --replace \
       'os.path.join(lib_dir, lib_name)' 'os.path.join("${tiledb}/lib", lib_name)'
-    
+
     # Disable failing test
     substituteInPlace tiledb/tests/test_examples.py --replace \
       "test_docs" "dont_test_docs"
+    # these tests don't always fail
+    substituteInPlace tiledb/tests/test_libtiledb.py --replace \
+      "test_varlen_write_int_subarray" "dont_test_varlen_write_int_subarray"
+    substituteInPlace tiledb/tests/test_metadata.py --replace \
+      "test_metadata_consecutive" "dont_test_metadata_consecutive"
   '';
 
   checkPhase = ''

